core/loader/json-feed.ts

```diff
@@ -0,0 +1,137 @@
+import type { TextResponse } from '../download.js'
+import type { OriginPost } from '../post.js'
+import { createPostsPage } from '../posts-page.js'
+import type { Loader } from './index.js'
+import { findAnchorHrefs, findLinksByType, toTime } from './utils.js'
+
+// https://www.jsonfeed.org/version/1.1/
+export type Author = {
+  avatar?: string
+  name?: string
+  url?: string
+}
+
+export type Item = {
+  /** deprecated from 1.1 version */
+  author?: Author
+  authors?: Author[]
+  banner_image?: string
+  content_html?: string
+  content_text?: string
+  date_modified?: string
+  date_published?: string
+  external_url?: string
+  id: string
+  image?: string
+  summary?: string
+  tags?: string[]
+  title?: string
+  url?: string
+}
+
+export type JsonFeed = {
+  /** deprecated from 1.1 version */
+  author?: Author
+  authors?: Author[]
+  description?: string
+  favicon?: string
+  feed_url?: string
+  home_page_url?: string
+  icon?: string
+  items: Item[]
+  next_url?: string
+  title: string
+  user_comment?: string
+  version: string
+}
+
+type ValidationRules = {
+  [key: string]: (value: unknown) => boolean
+}
+
+function isObjValid<ValidatedType>(
+  obj: unknown,
+  rules: ValidationRules
+): obj is ValidatedType {
+  if (typeof obj !== 'object' || obj === null || Array.isArray(obj)) {
+    return false
+  }
+
+  let objRecord = obj as Record<string, unknown>
+
+  for (let field in rules) {
+    if (!(field in objRecord) || !rules[field]!(objRecord[field])) {
+      // eslint-disable-next-line no-console
+      console.error(
+        `Value ${objRecord[field]} of object field ${field} is not valid`
```

Comment:
If `objRecord[field]` is not a simple string it will be printed as `[Object]`

We can simplify debugging a little by using:

```js
console.error(
    `JSON feed field ${field} is not valid`, objRecord[field]
}
```

Also let’s clarify that it is JSON feed validation.